sympy/matrices/matrices.py

```diff
@@ -189,6 +189,9 @@ def __neg__(self):
         return -1*self
 
     def _matrix_pow_by_jordan_blocks(self, num):
+        if (num < 0) == True and self.det() == 0:
```

Comment:
The eigenvalues are considered [here](https://github.com/sympy/sympy/blob/master/sympy/matrices/matrices.py#L197). It would be possible to raise an exception at this point.

If the determinant is computed, I think it could be done before calling _matrix_pow_by_jordan_blocks.